### PR TITLE
Upgrade all the way to Rails 6.0.3.4

### LINKED ISSUE #80: Upgrade to Rails 6.0.3.3

### PR DESCRIPTION
**Pull request by t27duck**
Closes #80 

## The Problem

The app is on Rails 5.1.x which is no longer supported. It was asked that the app be up to date with Rails 6.0.x.

## The Solution

Over multiple commits, the app has been stepped through "upgrading" from the Rails 5.1.x it was on, to the last 5.1.x, to the last 5.2.x, and then to the latest 6.0.x.

A lot of gems had to be updated to accommodate the update.

At least on my system, specs pass, cucumber tests "function" (see my PR comment in the Gemfile), and the local dev server renders the app. Was able to log in as the local admin and "approve" job postings.

The app itself could probably use a lot more love (like, remove coffescript gem, maybe go to webpacker, stop with cucumber, revisit and update the config files, etc etc), but this at least gets it to a current Rails?

### CODE REVIEW

File: Gemfile
Comment (t27duck) [inline Gemfile:60]: A lot of muddy work regarding cucumber, capybara, and related gems here... there's actually a downgrade to one of them in this change set. Due to major gherkin changes along with capybara matchers and other "stuff", someone more knowledgeable than I would need to more or less audit and/or rewrite the feature tests. For now, this works, minus what SHOULD be a safe "error" where minitest is invoked accidentally and complains about an unknown `--format` flag.

File: Gemfile
Comment (t27duck) [inline Gemfile:11]: The `sass` gem is dead. `sassc` is the last hold out and needed for what's left of asset pipeline.

File: app/assets/javascripts/application.js
Comment (t27duck) [inline app/assets/javascripts/application.js:6]: Sometime asset pipeline apparently was updated so it doesn't like files that end in "source"? Changing this filename made it suddenly find it again in current versions of the gem.

File: app/controllers/admin/jobs_controller.rb
Comment (t27duck) [inline app/controllers/admin/jobs_controller.rb:31]: So `company` was never here. Meaning old Rails was either letting it in or `company` was never being set for the records?

File: app/models/job.rb
Comment (t27duck) [inline app/models/job.rb:42]: I don't know what the "slugged" gem is or why a full on gem was needed for pretty much a `to_param` replacement. All I know is, I hate it.

Anyway, these dynamic methods are no longer in Rails and have replacements. But, the slugged gem hasn't been touched in six years so... yeah.

File: config/application.rb
Comment (t27duck) [inline config/application.rb:30]: This is the same compatibility-wise. Rails more or less just absorbed the adapter.

File: config/environments/production.rb
Comment (t27duck) [inline config/environments/production.rb:66]: Current day `I18n` installs warn you that this needs to change or things "may" break depending on how much this app relies on localizations.

File: config/initializers/assets.rb
Comment (t27duck) [inline config/initializers/assets.rb:1]: All I know is, current asset pipeline wouldn't work without this.

File: config/initializers/secret_token.rb
Comment (t27duck) [inline config/initializers/secret_token.rb:2]: This may or may not work in current Rails? Usually it looks for a `RAILS_SECRET_KEY_BASE` or some environment variable named similarly. I **think** this will still work.

File: config/initializers/secret_token.rb
Comment (t27duck) [inline config/initializers/secret_token.rb:4]: Technically not needed anymore. Rails 6 will set the secret key for you in dev and test mode. But, I suppose out of scope for this PR.

File: config/newrelic.yml
Comment (t27duck) [inline config/newrelic.yml:1]: Uh, yeah. Did this ever work?

File: spec/factories.rb
Comment (t27duck) [inline spec/factories.rb:3]: Factories require the attributes be in callable blocks from now on.

**Comment by t27duck** [comment]
Updated to the most recent Rails that was released today. Should be good.

File: app/models/job.rb
Comment (rreinhardt9) [reply app/models/job.rb:42]: Oh nice!!! Last time I looked into this the "slugged" gem is where I got stuck. like you said, it hadn't been updated in years. I think it's being used to create the slugs in the links for jobs. Love seeing how you got passed that part!

**Comment by t27duck** [comment]
Meant to do this sooner... per the discussion in the slack, I'll ping @armilam as a backup eventual looker-overer if someone else can't get to it.

File: app/models/job.rb
Comment (wuz) [reply app/models/job.rb:42]: Been looking at swapping the slugged gem out for something like [`friendly_id`](https://github.com/norman/friendly_id) here. Might be something to add as a new issue :D

File: app/controllers/admin/jobs_controller.rb
Comment (wuz) [reply app/controllers/admin/jobs_controller.rb:31]: Gonna guess the former since we use the company name in the job titles.

File: app/models/redirect.rb
Comment (wuz) [inline app/models/redirect.rb:15]: I'm not positive what we do with "visits", but I'm wondering if it might be beneficial to move this over to a real analytics platform? Maybe something self-hosted and not GA? @mileszs and @armilam what do you think?

File: features/step_definitions/admin_steps.rb
Comment (wuz) [inline features/step_definitions/admin_steps.rb:2]: Oh thank god we are using FactoryBot now lol

**Review by wuz — APPROVED** [review]
This looks pretty great to me, but I'll let @armilam weigh in before we merge the change! Thanks so much for doing this!

File: app/models/redirect.rb
Comment (armilam) [reply app/models/redirect.rb:15]: I have no idea what visits is here. I'm not sure what we want to do with visit data, but if there is a purpose, then I agree a real platform might be in order if we can get it free or super cheap.

**Comment by armilam** [comment]
After scanning the code the changes seem fine. I trust you and @wuz's review. I'm just going to run this locally and do some smoke testing real quick.

**Comment by armilam** [comment]
http://localhost:3000/about

Screen Shot 2020-10-08 at 2 33 52 PM

**Comment by t27duck** [comment]
Can you guess what page I didn't click on? 🙃 

I'll see what I can do.

**Comment by armilam** [comment]
Thanks! Besides that one thing, it appears everything works. Thank you for getting us up to date!

**Comment by t27duck** [comment]
@armilam The about us page should be "fixed." By "fixed" I mean, "The board member images were slip between `public/images` and `app/assets/images` so I took the easy way out and moved them all out of asset pipeline to `public/images/board` (for organizational purposes), adjusted the `image_tag` and it all worked locally for me."

I did an `ag` on the codebase to make sure those images weren't referenced anywhere else. Looks like they weren't.

**Comment by armilam** [comment]
It looks like everything is working. @t27duck thank you for doing this - I really appreciate it!

Now to test in production. :yolo:

**Pull request merged**
